app/scripts/controllers/preferences.js

```diff
@@ -51,7 +51,11 @@ export default class PreferencesController {
       useNonceField: false,
       usePhishDetect: true,
       dismissSeedBackUpReminder: false,
-      useStaticTokenList: false,
+
+      // contradictory to the name the useStaticTokenList
+      // set to true means the dynamic list from the API is being used
+      // set to false will be using the static list from contract-metadata
+      useStaticTokenList: true,
```

Comment:
I've mentioned this in a previous comment but I really think we should switch this around so that useStaticTokenList actually corresponds to using the static token list rather than inverting it...

Reply:
The flag useStaticTokenList was added first in the controller repo and the default value was supposed to be `false`. But while adding the extension changes, I realized we need the toggle to be ON by default, hence the flag has to be `true`, after a discussion with @danjm we decided to not change the controller repo instead inverted the value we are sending from the extension.

I kept the same variable name to keep it consistent with the controller, now I am wondering if I should change that to `useTokenDetection`?

Reply:
Thanks for the background context @NiranjanaBinoy. I would vote yes to changing it to `useTokenDetection`!